exercises/atbash_cipher/example.cr

```diff
@@ -0,0 +1,13 @@
+module Atbash
+  MAPPING = (alp = ('a'..'z').to_a).zip(alp.reverse).to_h
+  extend self
+  def encode(message : String) : String
+    message = message.downcase.gsub(/[^a-z0-9]/, "")
+    encoded_chars = message.chars.map{|c| MAPPING[c]? || c}
+    encoded_string = ""
+    encoded_chars.each_with_index do |c, i|
+      encoded_string += ((i%5 == 0 ? " " : "") + c)
+    end
+    encoded_string.strip
+  end
```

Comment:
I agree, although the Ruby implement almost makes the fairly simple problem
look complex by creating functions the do little more than call built in
Ruby functions. I'll go through and add a couple where it makes sense.

On Jul 21, 2016 1:00 AM, "Max Helmetag" <notifications@github.com> wrote:

> In exercises/atbash_cipher/example.cr
> <https://github.com/exercism/xcrystal/pull/19#discussion_r71655677>:
>
> > @@ -0,0 +1,13 @@
> > +module Atbash
> > +  MAPPING = (alp = ('a'..'z').to_a).zip(alp.reverse).to_h
> > +  extend self
> > +  def encode(message : String) : String
> > +    message = message.downcase.gsub(/[^a-z0-9]/, "")
> > +    encoded_chars = message.chars.map{|c| MAPPING[c]? || c}
> > +    encoded_string = ""
> > +    encoded_chars.each_with_index do |c, i|
> > +      encoded_string += ((i%5 == 0 ? " " : "") + c)
> > +    end
> > +    encoded_string.strip
> > +  end
>
> It might be worth splitting up the individual functions into their own
> methods and maybe putting this into a class. Up to you... but I think it
> would be cleaner.
>
> Like:
>
> def encoded_chars(message)
>   message.chars.map{|c| MAPPING[c]? || c}end
>
> Or something.
>
> Just looking at the Ruby implementation...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/exercism/xcrystal/pull/19/files/63ca362bf5754190e9bb96b45a7eaccd0afc90c0#r71655677>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHmpPDgU9yLWbP2-1Wutq4VfoPmLhNGks5qXxkVgaJpZM4JRIaI>
> .
>